File: 0198-house-robber/0198-house-robber.java
class Solution {
    public int rob(int[] nums) {
        final int n = nums.length;
        int[] dp = new int[n];
        dp[0] = nums[0];

        for(int i = 1; i < n; i++) {
            dp[i] = Math.max(nums[i] + ( i>= 2 ?  dp[i-2] : 0), dp[i-1]);
        }
        return dp[n-1];
    }
}

File: 0198-house-robber/0198-house-robber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var rob = function(nums) {
    
    const n = nums.length;
    
    let dp = new Array(n);
    
    dp[0] = nums[0];
    dp[1] = Math.max(nums[0], nums[1]);
    
    for(let i = 2; i < n; i++) {
        dp[i] = Math.max(nums[i] + dp[i-2], dp[i-1]);
    }
    return dp[n-1];
};